FinSearchUnified/Bundle/SearchBundleFindologic/QueryBuilder.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic;
+
+use Exception;
+use Shopware\Bundle\PluginInstallerBundle\Service\InstallerService;
+use Shopware\Components\HttpClient\HttpClientInterface;
+use Shopware\Components\HttpClient\RequestException;
+use Shopware\Components\HttpClient\Response;
+use Shopware_Components_Config;
+
+class QueryBuilder
+{
+    const BASE_URL = 'https://service.findologic.com/ps/xml_2.0/';
+    const ALIVE_ENDPOINT = 'alivetest.php';
+    const SEARCH_ENDPOINT = 'index.php';
+    const NAVIGATION_ENDPOINT = 'selector.php';
+
+    /**
+     * @var HttpClientInterface
+     */
+    protected $httpClient;
+
+    private $parameters;
+
+    private $shopUrl;
+
+    private $shopKey;
+
+    /**
+     * QueryBuilder constructor.
+     *
+     * @param HttpClientInterface $httpClient
+     * @param InstallerService $installerService
+     * @param Shopware_Components_Config $config
+     *
+     * @throws Exception
+     */
+    public function __construct(
+        HttpClientInterface $httpClient,
+        InstallerService $installerService,
+        Shopware_Components_Config $config
+    ) {
+        $plugin = $installerService->getPluginByName('FinSearchUnified');
+
+        $this->parameters = [];
+        $this->httpClient = $httpClient;
+        $this->shopKey = $config->offsetGet('ShopKey');
+        $this->shopUrl = rtrim(Shopware()->Shop()->getHost(), '/') . '/';
+        $this->parameters = [
+            'userip' => $this->getClientIp(),
+            'revision' => $plugin->getVersion(),
+            'shopkey' => $this->shopKey
+        ];
+    }
+
+    /**
+     * @param bool $isSearch
+     *
+     * @return string|null
+     */
+    public function execute($isSearch = true)
+    {
+        if ($isSearch) {
+            $endpoint = self::SEARCH_ENDPOINT;
+        } else {
+            $endpoint = self::NAVIGATION_ENDPOINT;
+        }
+
+        $url = sprintf(
+            '%s%s%s?%s',
+            self::BASE_URL,
```

Comment:
* Remove the slashes at the end of `self::BASE_URL` and `$this->shopUrl`.
* Add the slashes to `sprintf` e.g. `sprintf('%s/%s/%s?%s' ,...)`.
* Apply the same changes to method `isAlive`.

This will make the `sprintf` more readable.